### PR TITLE
Add a Developers section w/ style guidelines to README

### PR DESCRIPTION
**Pull request by petroav**
This is the second half of the style guidelines discussed during the Boston Presto meetup (the other half is here https://github.com/airlift/codestyle/pull/3). I split the guidelines into two sections: one that is more general (to be merged in codestyle) and another section that is more specific to the Presto codebase (this PR).

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -71,3 +71,16 @@ Run a query to see the nodes in the cluster:
...
+
+## Developers
+
+We recommend you use IntelliJ as your IDE. The code style template for the project can be found in the [codestyle](https://github.com/airlift/codestyle) repository along with our general programming and Java guidelines. In addition to those you should also adhere to the following:
+
+* Alphabetize sections in the documentation source files (both in table of contents files and other regular documentation files). In general, alphabetize methods/variables/sections if such ordering already exists in the surrounding code.
+* Use Guava's `Splitter` for parsing strings.
+* When appropriate, use the Java 8 stream API. However, note that the stream implementation does not perform well so avoid using it in inner loops or otherwise performance sensitive sections.
+* Categorize errors when throwing exceptions. For example, PrestoException takes an error code as an argument, `PrestoException(HIVE_TOO_MANY_OPEN_PARTITIONS)`. This categorization lets you generate reports so you can monitor the frequency of various failures.
+* Ensure that all files have the appropriate license; you can generate the license by running `mvn license:format`.
```
Comment (petroav) [inline README.md]: There was a caveat here, I don't remember what it was.

File: README.md [hunk 2]
```diff
@@ -71,3 +71,16 @@ Run a query to see the nodes in the cluster:
...
+* Consider using String formatting (printf style formatting using the Java `Formatter class`): `String.format("Session property %s is invalid: %s", name, value)`. Sometimes, if you only need to append something, consider using the `+` operator.
+* Avoid using the ternary operator.
```
Comment (petroav) [inline README.md]: There was some discussion around this item as well but I didn't get a chance to write it down.

**Comment by petroav** [comment]
Addressed all of @electrum's comments and rebased.

**Comment by petroav** [comment]
I just realized that the `--force` I did to rebase erased @electrum's comments. Sorry about that. If you look at the diff of my second commit, it mirrors his suggestions exactly.

File: README.md [hunk 3]
```diff
@@ -71,3 +71,16 @@ Run a query to see the nodes in the cluster:
 In the sample configuration, the Hive connector is mounted in the `hive` catalog, so you can run the following queries to show the tables in the Hive database `default`:
 
     SHOW TABLES FROM hive.default;
+
+## Developers
+
+We recommend you use IntelliJ as your IDE. The code style template for the project can be found in the [codestyle](https://github.com/airlift/codestyle) repository along with our general programming and Java guidelines. In addition to those you should also adhere to the following:
+
+* Alphabetize sections in the documentation source files (both in table of contents files and other regular documentation files). In general, alphabetize methods/variables/sections if such ordering already exists in the surrounding code.
+* Use Guava's `Splitter` for parsing strings.
```
Comment (electrum) [inline README.md]: This seems rather specific.

**Comment by electrum** [comment]
@martint This looks fine to me, but I'll let you merge

**Comment by martint** [comment]
Merged, thanks!

**Pull request closed without merging**